x-pack/plugins/rule_registry/server/plugin.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { PluginInitializerContext, Plugin, CoreSetup } from 'src/core/server';
+import { PluginSetupContract as AlertingPluginSetupContract } from '../../alerting/server';
+import { RuleRegistry } from './rule_registry';
+import { defaultIlmPolicy } from './rule_registry/defaults/ilm_policy';
+import { defaultFieldMap } from './rule_registry/defaults/field_map';
+import { RuleRegistryConfig } from '.';
+
+export type RuleRegistryPluginSetupContract = RuleRegistry<typeof defaultFieldMap>;
+
+export class RuleRegistryPlugin implements Plugin<RuleRegistryPluginSetupContract> {
+  constructor(private readonly initContext: PluginInitializerContext) {
+    this.initContext = initContext;
+  }
+
+  public setup(
+    core: CoreSetup,
+    plugins: { alerting: AlertingPluginSetupContract }
+  ): RuleRegistryPluginSetupContract {
+    const globalConfig = this.initContext.config.legacy.get();
+    const config = this.initContext.config.get<RuleRegistryConfig>();
+
+    const logger = this.initContext.logger.get();
+
+    const rootRegistry = new RuleRegistry({
+      core,
+      ilmPolicy: defaultIlmPolicy,
+      fieldMap: defaultFieldMap,
+      kibanaIndex: globalConfig.kibana.index,
+      name: 'alert-history',
+      kibanaVersion: this.initContext.env.packageInfo.version,
+      logger: logger.get('root'),
+      alertingPluginSetupContract: plugins.alerting,
+      writeEnabled: config.writeEnabled,
+    });
+
+    return rootRegistry;
+  }
+
+  public start() {}
```

Comment:
No change now, but we may need to temporarily expose `alerting` via the `PluginStart` contract to aid in migrating the `security-solution` plugin to the rule-registry. It's currently a dependency of the [EndpointAppContextService](https://github.com/elastic/kibana/blob/9ceea45c1e3409a32fdf98c46aba269b9ec0060e/x-pack/plugins/security_solution/server/plugin.ts#L437). 

For testing purposes I went ahead and did that in the [reference rule branch](https://github.com/elastic/kibana/pull/96015/files#diff-acb9efe19130cb0a2d9b5a83c578f9d28ad265272746014c27f167f3882c90c3R47-R51) so I could test creating rules against the `rule-registry` within `security-solution`, but we can address that in a future PR if both end up needing to co-exist for a short period of time.

Reply:
Curious why you need it? the rule registry register all types with the alerting plugin, and the intent was to be fully compatible with the legacy alert types. I'd have to double check whether that is actually the case though.

Reply:
I believe it's just a temporary measure to enable a passthrough to alerting for the rules that aren't yet ported to rule-registry.

Reply:
> I believe it's just a temporary measure to enable a passthrough to alerting for the rules that aren't yet ported to rule-registry.

For rule registration we only need `registerType()` off of the `PluginSetupContract` which is exposed via [ruleRegistry.alertingPluginSetupContract](https://github.com/elastic/kibana/pull/96015/files#diff-d1c12442b96c85436a1774d602c09e6c7df7daf8728177e7e7883a7ecf238e5cR293-R294), so we're all good there. 👍 

This is necessary as the `fleet` side of `security-solution` needs `getAlertsClientWithRequest()` off the `PluginStartContract` for installing the pre-packaged rules when you enable the `Elastic Security` integration over in fleet-land.

https://github.com/elastic/kibana/blob/f21417660af7dfc65bea05bf85253bb96c837c87/x-pack/plugins/security_solution/server/fleet_integration/handlers/install_prepackaged_rules.ts#L70


That said, @rw-access is working to enable the installation of pre-packaged rules via a dedicated fleet package, so we might be able to work around the dependency, but will most likely need it through 7.14/15. There is also the larger question of what does it mean to install pre-packaged rules generally, so we should have that conversation too 🙂